contracts/contracts/GrantRoundManager.sol

```diff
@@ -63,27 +57,27 @@ contract GrantRoundManager {
   // --- Constructor ---
   constructor(
     GrantRegistry _registry,
-    ISwapRouter _router,
-    IERC20 _donationToken
-  ) {
+    IERC20 _donationToken,
+    address _factory,
+    address _WETH9
+  ) SwapRouter(_factory, _WETH9) {
     // Validation
     require(_registry.grantCount() >= 0, "GrantRoundManager: Invalid registry");
-    require(address(_router).isContract(), "GrantRoundManager: Invalid router"); // Router interface doesn't have a state variable to check
     require(_donationToken.totalSupply() > 0, "GrantRoundManager: Invalid token");
 
     // Set state
     registry = _registry;
-    router = _router;
+    WETH = IERC20(_WETH9);
     donationToken = _donationToken;
 
     // Token approvals of common tokens
     // TODO inherit from SwapRouter to remove the need for this approvals and extra safeTransferFrom before swap
-    IERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F).safeApprove(address(_router), type(uint256).max); // DAI
-    IERC20(0xDe30da39c46104798bB5aA3fe8B9e0e1F348163F).safeApprove(address(_router), type(uint256).max); // GTC
-    IERC20(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48).safeApprove(address(_router), type(uint256).max); // USDC
-    IERC20(0xdAC17F958D2ee523a2206206994597C13D831ec7).safeApprove(address(_router), type(uint256).max); // USDT
-    IERC20(0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599).safeApprove(address(_router), type(uint256).max); // WBTC
-    WETH.safeApprove(address(_router), type(uint256).max); // WETH
+    IERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F).safeApprove(address(this), type(uint256).max); // DAI
+    IERC20(0xDe30da39c46104798bB5aA3fe8B9e0e1F348163F).safeApprove(address(this), type(uint256).max); // GTC
+    IERC20(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48).safeApprove(address(this), type(uint256).max); // USDC
+    IERC20(0xdAC17F958D2ee523a2206206994597C13D831ec7).safeApprove(address(this), type(uint256).max); // USDT
+    IERC20(0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599).safeApprove(address(this), type(uint256).max); // WBTC
+    WETH.safeApprove(address(this), type(uint256).max); // WETH
```

Comment:
We can get rid of all these approvals (and the comments above them). These were only needed because we had to approve the router to spend this contract's tokens, but now this contract is the router so this is not required